File: DoorsPOS.WinApp/CS/frmManualInv.cs
// VBConversions Note: VB project level imports
using System.Data;
using System.Collections.Generic;
using PDSA.Data;
using PDSA.WinForms.Forms;
using PDSA.Security;
using System.Diagnostics;
using PDSA.Common;
using PDSA.WinForms;
using PDSA.WinForms.Security;
using PDSA.PDSACryptography;
using DoorsPOS.Utility;
using Microsoft.VisualBasic;
using System.Collections;
using System;
using DoorsPOS.BusinessLayer;
using System.Windows.Forms;
using System.Drawing;
using System.Linq;
using DoorsPOS.DataLayer;
// End of VB project level imports

using PDSA.DataLayer.DataClasses;
using PDSA.Validation;
using DevExpress.XtraEditors;


namespace DoorsPOS.WinApp
{
	public partial class frmManualInv
	{
		public frmManualInv()
		{
			InitializeComponent();
		}
		stocksManager mgr9;
		
		public void btnItemSearch_Click(object sender, EventArgs e)
		{
			frmItemSearch frmitemserch = new frmItemSearch();
			frmitemserch.ShowDialog();
			txtItem.Text = string.Empty;
			txtStckId.Text = string.Empty;
			txtItem.Text = frmitemserch.vItem.ToString();
			txtStckId.Text = frmitemserch.vStckid.ToString();
			if (txtStckId.Text == string.Empty)
			{
				return;
			}
			else
			{
				ceQty.Focus();
			}
			ceQty.Focus();
		}
		
		public void frmManualInv_FormClosed(object sender, FormClosedEventArgs e)
		{
			this.Dispose();
		}
		public void frmManualInv_FormClosing(object sender, FormClosingEventArgs e)
		{
			if (btnSave.Enabled == true)
			{
				e.Cancel = true;
				MessageBox.Show("Save or Cancel the Entry First Before Closing the Form.");
			}
		}
		
		public void btnSave_Click(object sender, EventArgs e)
		{
			if (txtItem.Text == string.Empty)
			{
				MessageBox.Show("Please select the item counted.");
				txtItem.Focus();
				return;
			}
			
			if (ceQty.Value < 0)
			{
				MessageBox.Show("Please enter the number of items counted.");
				ceQty.Focus();
				return;
			}
			
			if (deBoDate.Text == string.Empty)
			{
				MessageBox.Show("Please enter the date the item was counted.");
				deBoDate.Focus();
				return;
			}
			
			
			PDSATransaction TranMI = new PDSATransaction();
			InvntoryManager mgrMI = new InvntoryManager();
			
			try
			{
				mgrMI.DataObject.TransactionType = PDSATransactionType.Insert;
				mgrMI.DataObject.Entity.amnt = 0;
				//mgrMI.DataObject.Entity.amnt = 0
				mgrMI.DataObject.Entity.prevavlbl = 0; //frmItemSearch.vAvlbl
				mgrMI.DataObject.Entity.midate = DateTime.Parse(deBoDate.Text);
				mgrMI.DataObject.Entity.qty = (int) ceQty.Value;
				mgrMI.DataObject.Entity.sInsertid = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrMI.DataObject.Entity.sLastUpdateid = System.Convert.ToString(PDSAAppConfig.CurrentLoginID);
				mgrMI.DataObject.Entity.stckid = int.Parse(txtStckId.Text.Trim());
				
				TranMI.Add(mgrMI.DataObject);
				
				spInvntoryManager mgrspInv = new spInvntoryManager();
				mgrspInv.Entity.StckIid = int.Parse(txtStckId.Text);
				TranMI.Add(mgrspInv.DataObject);
				
				TranMI.Execute();
				DisableControls();
				MessageBox.Show("Manual Inventory Entry Successfully Saved", "Saved");
				InitControls();
			}
			catch (PDSAValidationException ex)
			{
				MessageBox.Show(ex.Message);
				return;
			}
			catch (Exception ex)
			{
				MessageBox.Show(ex.ToString());
				return;
			}
		}
		public void btnCancel_Click(object sender, EventArgs e)
		{
			DisableControls();
			InitControls();
		}
		
		public void btnNew_Click(object sender, EventArgs e)
		{
			EnableControls();
			InitControls();
		}
		
		public void ceQty_KeyPress(object sender, KeyPressEventArgs e)
		{
			if (e.KeyChar == '\r')
			{
				btnSave.Focus();
			}
		}
		
		private void EnableControls()
		{
			btnSave.Enabled = true;
			btnCancel.Enabled = true;
			GroupControl1.Enabled = true;
			btnClose.Enabled = false;
			btnNew.Enabled = false;
			deBoDate.Focus();
		}
		
		private void DisableControls()
		{
			btnSave.Enabled = false;
			btnCancel.Enabled = false;
			GroupControl1.Enabled = false;
			btnClose.Enabled = true;
			btnNew.Enabled = true;
			btnNew.Focus();
		}
		
		private void InitControls()
		{
			deBoDate.EditValue = DateAndTime.Today;
			txtItem.Text = "";
			txtStckId.Text = "";
			ceQty.Value = 0;
		}
		
		public void btnClose_Click(object sender, EventArgs e)
		{
			this.Close();
		}
		
	}
}

File: DoorsPOS.WinApp/frmManualInv.vb
Imports PDSA.DataLayer.DataClasses
Imports PDSA.Validation
Imports DevExpress.XtraEditors

Public Class frmManualInv
    Dim mgr9 As stocksManager

    Private Sub btnItemSearch_Click(sender As Object, e As EventArgs) Handles btnItemSearch.Click
        Dim frmitemserch As New frmItemSearch
        frmitemserch.ShowDialog()
        txtItem.Text = String.Empty
        txtStckId.Text = String.Empty
        txtItem.Text = frmitemserch.vItem.ToString
        txtStckId.Text = frmitemserch.vStckid.ToString
        If txtStckId.Text = String.Empty Then
            Exit Sub
        Else
            ceQty.Focus()
        End If
        ceQty.Focus()
    End Sub

    Private Sub frmManualInv_FormClosed(sender As Object, e As FormClosedEventArgs) Handles Me.FormClosed
        Me.Dispose()
    End Sub
    Private Sub frmManualInv_FormClosing(sender As Object, e As FormClosingEventArgs) Handles Me.FormClosing
        If btnSave.Enabled = True Then
            e.Cancel = True
            MessageBox.Show("Save or Cancel the Entry First Before Closing the Form.")
        End If
    End Sub

    Private Sub btnSave_Click(sender As Object, e As EventArgs) Handles btnSave.Click
        If txtItem.Text = String.Empty Then
            MessageBox.Show("Please select the item counted.")
            txtItem.Focus()
            Exit Sub
        End If

        If ceQty.Value < 0 Then
            MessageBox.Show("Please enter the number of items counted.")
            ceQty.Focus()
            Exit Sub
        End If

        If deBoDate.Text = String.Empty Then
            MessageBox.Show("Please enter the date the item was counted.")
            deBoDate.Focus()
            Exit Sub
        End If


        Dim TranMI As PDSATransaction = New PDSATransaction()
        Dim mgrMI As InvntoryManager = New InvntoryManager()

        Try
            mgrMI.DataObject.TransactionType = PDSATransactionType.Insert
            mgrMI.DataObject.Entity.amnt = 0
            'mgrMI.DataObject.Entity.amnt = 0
            mgrMI.DataObject.Entity.prevavlbl = 0 'frmItemSearch.vAvlbl
            mgrMI.DataObject.Entity.midate = CDate(deBoDate.Text)
            mgrMI.DataObject.Entity.qty = CInt(ceQty.Value)
            mgrMI.DataObject.Entity.sInsertid = PDSAAppConfig.CurrentLoginID
            mgrMI.DataObject.Entity.sLastUpdateid = PDSAAppConfig.CurrentLoginID
            mgrMI.DataObject.Entity.stckid = CInt(Trim(txtStckId.Text))

            TranMI.Add(mgrMI.DataObject)

            Dim mgrspInv As spInvntoryManager = New spInvntoryManager()
            mgrspInv.Entity.StckIid = CInt(txtStckId.Text)
            TranMI.Add(mgrspInv.DataObject)

            TranMI.Execute()
            DisableControls()
            MessageBox.Show("Manual Inventory Entry Successfully Saved", "Saved")
            InitControls()
        Catch ex As PDSAValidationException
            MessageBox.Show(ex.Message)
            Exit Sub
        Catch ex As Exception
            MessageBox.Show(ex.ToString())
            Exit Sub
        End Try
    End Sub
    Private Sub btnCancel_Click(sender As Object, e As EventArgs) Handles btnCancel.Click
        DisableControls()
        InitControls()
    End Sub

    Private Sub btnNew_Click(sender As Object, e As EventArgs) Handles btnNew.Click
        EnableControls()
        InitControls()
    End Sub

    Private Sub ceQty_KeyPress(sender As Object, e As KeyPressEventArgs) Handles ceQty.KeyPress
        If e.KeyChar = Chr(13) Then
            btnSave.Focus()
        End If
    End Sub

    Private Sub EnableControls()
        btnSave.Enabled = True
        btnCancel.Enabled = True
        GroupControl1.Enabled = True
        btnClose.Enabled = False
        btnNew.Enabled = False
        deBoDate.Focus()
    End Sub

    Private Sub DisableControls()
        btnSave.Enabled = False
        btnCancel.Enabled = False
        GroupControl1.Enabled = False
        btnClose.Enabled = True
        btnNew.Enabled = True
        btnNew.Focus()
    End Sub

    Private Sub InitControls()
        deBoDate.EditValue = DateAndTime.Today
        txtItem.Text = ""
        txtStckId.Text = ""
        ceQty.Value = 0
    End Sub

    Private Sub btnClose_Click(sender As Object, e As EventArgs) Handles btnClose.Click
        Me.Close()
    End Sub

End Class